whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/FullTextSearch.java

```diff
@@ -164,23 +173,24 @@ protected SearchResponse doSearch(final IndexReader indexReader, final TaxonomyR
 
                     indexSearcher.search(query, MultiCollector.wrap(topFieldCollector, facetsCollector));
 
-                    final List<Document> documents = Lists.newArrayList();
+                    final Map<RpslObject, Document> rpslObjectToDocument = Maps.newHashMap();
```

Comment:
I would flip this Map around (document with a matching RpslObject) or create a list of "pairs"

Reply:
I would rely on Rpslobject as key  rather then luecene document. As Rpslobject have our own implementation of  equals and hashcode

Reply:
Good point, thanks!